src/runtime_src/core/pcie/tools/xbmgmt/xmc.cpp

```diff
@@ -500,15 +507,52 @@ int XMC_Flasher::writeReg(unsigned RegOffset, unsigned value) {
     return 0;
 }
 
+static std::string getStatus(int status, std::map<int, std::string> &map)
+{
+    auto entry = map.find(status);
+    std::ostringstream os;
+
+    os << std::hex << entry->first;
+
+    if (entry != map.end())
+	    os << "(" << entry->second << ")";
+
+    return os.str();
```

Comment:
Use boost format:
`return entry != map.end() ? boost::str(boost::format("%x (%s)") % entry->first % entry->second) : "";`

Reply:
There is a bug, I fixed. Code should be:
os << std::hex << status;
if (entry ...)
So that we can still display any status which is not defined in the map.

If we use boost, the code would be:
if (entry != map.end()) {
           boost::str(boost::format("%x (%s)") % entry->first % entry->second)
} else {
           boost::str(boost::format("%x") % status)
}  

Seems standard std::cout is cosine.